monkey/monkey_island/cc/services/agent_plugin_service/flask_resources/installed_agent_plugins_manifests.py

```diff
@@ -0,0 +1,58 @@
+import logging
+from http import HTTPStatus
+from typing import Any, Dict
+
+from flask import make_response
+from flask_security import auth_token_required, roles_accepted
+
+from common.agent_plugins import AgentPluginManifest, AgentPluginType
+from monkey_island.cc.flask_utils import AbstractResource
+from monkey_island.cc.repositories import UnknownRecordError
+from monkey_island.cc.services.authentication_service import AccountRole
+
+from .. import IAgentPluginService
+
+logger = logging.getLogger(__name__)
+
+
+class InstalledAgentPluginsManifests(AbstractResource):
+    urls = ["/api/agent-plugins/installed/manifests"]
+
+    def __init__(self, agent_plugin_service: IAgentPluginService):
+        self._agent_plugin_service = agent_plugin_service
+
+    @auth_token_required
+    @roles_accepted(AccountRole.ISLAND_INTERFACE.name)
+    def get(self):
+        """
+        Get manifests of all installed plugins
+        """
+
+        try:
+            installed_agent_plugins_manifests = (
+                self._agent_plugin_service.get_all_plugin_manifests()
+            )
+            installed_agent_plugins_manifests_simplified = (
+                self._get_simplified_installed_agent_plugins_manifests(
+                    installed_agent_plugins_manifests
+                )
+            )
+
+            return make_response(installed_agent_plugins_manifests_simplified, HTTPStatus.OK)
+        except UnknownRecordError:
```

Comment:
`get_all_plugin_manifests()` does not raise `UnknownRecordError`. It can raise `RetrievalError`. Flask should catch this and return a 500, so we don't need to.